src/compute/src/server.rs

```diff
@@ -294,7 +296,10 @@ pub async fn compute_node_serve(
         system_params_manager.watch_params(),
     ));
 
-    let heap_profiler = HeapProfiler::new(total_memory_bytes, config.server.heap_profiling.clone());
+    let heap_profiler = HeapProfiler::new(
+        opts.total_memory_bytes,
+        config.server.heap_profiling.clone(),
+    );
```

Comment:
This was set to compute_memory_bytes + storage_memory_bytes because the cn should not use memory more than that in normal case.

Reply:
I don't think so. 

The key point here is that the passed-in threshold is compared with jemalloc's allocated memory statistics. The memory that used by neither compute_memory_bytes nor storage_memory_bytes, what we called "reserved memory", is also counted in. 

In other words, as a fair comparison, either of the following is reasonable:

- Current heap memory v.s. 90% of total heap memory
- Current compute + storage v.s. 90% of total compute + storage

But now it's comparing "Current heap memory" v.s. "90% of total compute + storage", which is not reasonable.

Reply:
I was thinking Current heap memory v.s. all total heap memory.
But Current heap memory v.s. 90% of total heap size lgtm.